code/game/turfs/open.dm

```diff
@@ -216,6 +217,33 @@
 		add_overlay(wet_overlay)
 	HandleWet()
 
+/turf/open/proc/UpdateSlip()
+	switch(wet)
+		if(TURF_WET_WATER)
+			AddComponent(/datum/component/slippery, 60, NO_SLIP_WHEN_WALKING)
+		if(TURF_WET_LUBE)
+			AddComponent(/datum/component/slippery, 80, SLIDE | GALOSHES_DONT_HELP)
+		if(TURF_WET_ICE)
+			AddComponent(/datum/component/slippery, 120, SLIDE | GALOSHES_DONT_HELP)
+		if(TURF_WET_PERMAFROST)
+			AddComponent(/datum/component/slippery, 120, SLIDE_ICE | GALOSHES_DONT_HELP)
+		if(TURF_WET_SLIDE)
+			AddComponent(/datum/component/slippery, 80, SLIDE | GALOSHES_DONT_HELP)
+		else
+			qdel(GetComponent(/datum/component/slippery))
+
+/turf/open/ComponentActivated(datum/component/C)
+	..()
+	var/datum/component/slippery/S = C
+	if(!istype(S))
+		return
+	var/mob/living/L = S.slip_victim
+	switch(wet)
+		if(TURF_WET_WATER)
+			L.inertia_dir = 0
```

Comment:
What's the point of this anyway, you can't slip in no gravity from turfs can you?